### ISSUE TITLE
Issue creating vlan

### ISSUE THREAD

**Issue by Ari-E-S** [softlayer/terraform-provider-softlayer#40]
I'm having issues creating my first VLAN (never tried it before)

The plan is build without issues but, when trying to apply, an error comes up.

This code:

```
resource "softlayer_vlan" "vlan_wdc01_priv" {
   name = "vlan-wdc01-priv"
   datacenter = "wdc01"
   type = "PRIVATE"
   subnet_size = 64
}
```

Gets me this error:

```
* softlayer_vlan.vlan_wdc01_priv: Error creating vlan: reading body error: type mismatch - can't unmarshal slice to struct
```

I checked the data types for each field and they seem to match. What am I missing?

I'm already using v1.0.1

**Comment by renier** [comment]
@arielsalvo Probably another issue with xmlrpc. Duplicate of #38 most likely.

you can comment out the endpoint from your ~/.softlayer file for now as a workaround.

**Comment by Ari-E-S** [comment]
That changed the error to this:

```
* softlayer_vlan.vlan_wdc01_priv: Error during creation of vlan: SoftLayer_Exception_Public: At the moment this account is not permitted to order additional VLANs. To request an additional VLAN, please contact our sales team for assistance (HTTP 500)
```

This is a new account so no devices or vlans or anything has been created yet.
Is it normal?

**Comment by renier** [comment]
@arielsalvo That new error actually says what the problem is. And yes, that can happen in an account. You should contact support about it.

Can you create a VLAN through the portal?

**Comment by Ari-E-S** [comment]
I created the machines with terraform which, in turn, created the vlans (had to get these out to deliver)
I then picked up the created vlans with a datasource.

I opened a ticket with SL support but it didn't get me an appropriate answer so we'll escalate it.

I'll let you know if I get news.

in any case, should we leave this one open to work on the xmlrpc bug or join it with #38?

**Comment by renier** [comment]
@arielsalvo As far as I know, VLANs don't get created as a side effect of ordering VMs. Your VMs get assigned to pre-existing VLANs in the account.

I'm leaving this open for now.

**Comment by Ari-E-S** [comment]
@renier the very first VLAN (and subnet) of each kind (public/private) gets created with the first machine on the account.
I had to create one first because I need to specify the VLAN id in the definition of the 2 machines I'm creating so they are both added to the same VLANs.

**Comment by renier** [comment]
@arielsalvo Ok, I see. You can also deal with that in the same terraform run by using `depends_on` on the second and subsequent VM resource. They would all depend on the first one being created first, and then they can refer to the vlan ids it got.

**Comment by renier** [comment]
Actually, you should not need to use depends_on if you refer to another resource. Terraform should understand the dependency graph automatically.

**Comment by renier** [comment]
Fixed in 6c5defa68fe4e5fde5e6777a64845b198c77123c and bb77474d9e97ebbf1b08d2c51e313a5f95f319a1
